test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
+    std::vector<Source> sources;
+    Group append_group = block.createGroup("ag", "test");
+    Group set_group = block.createGroup("sg", "test");
+
+    for (auto it = names.begin(); it != names.end(); it++) {
+        sources.push_back(block.createSource(*it, "channel"));
+    }
+
+    // append each source in order
+    for (auto src : block.sources()) {
+        append_group.addSource(src);
+    }
+
+    // use the vector setter
+    set_group.sources(sources);
+
+    for (size_t idx = 0; idx < sources.size(); ++idx) {
+        CPPUNIT_ASSERT(sources[idx].name() == append_group.getSource(idx).name());
```

Comment:
`CPPUNIT_ASSERT_EQUAL(expected, actual)` gives nicer errors in case of broken tests. Still ok.